pieces/simple_csv/README.md

```diff
@@ -171,3 +171,19 @@ test tests::test_write_csv ... ok
 
 test result: ok. 3 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
 ```
+
+## Takeaways
+
+Notice in `read_csv` in the Rust file the return type is `Result<Vec<CityTemps>, Box<dyn std::error::Error>>`.
+In Python exceptions can be raised either explicitly or silently at any time, however this is not
+the case in Rust. If a function can error it has to return the `Result` type, specifying the type
+for success, the `Ok` value, and the type for errors, the `Err`. Any time a function is called
+that returns a `Result` both the `Ok` and the `Err` possiblities have to be handled. The `?` operator
+can be used when a function returns a `Result`, and specfifies that if the result of the function
+call is an error that error should be returned, otherwise use the `Ok` value.
+
+You will also notice that `expect` and `unwrap` are used. These are similiar to using `?`, but can
```

Comment:
Does it make sense to explain why `expect` is used in certain cases over `unwrap`? The Rust book says that in a single program where many areas where `unwrap` is used, it's hard to diagnose what exactly caused a panic, but with `expect`, the error message can be more explicit and tell the user what caused it (for example, invalid regex match).

Reply:
Yes, this makes sense. This is another area I probably cut too much out.